### PR TITLE
[docs] improve format of manpage

### PR DESCRIPTION
**Pull request by iw0nderhow**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Cleaning up the docs as specified in #1448.

Tasks completed:

- [x] Badges are unnecessary and unreadable -> removed
- [x] Description is missing header -> header added
- [x] Table of contents unnecessary -> removed
- [x] `INSTALLATION` unnecessary -> moved to bottom
- [x] `NEW FEATURES` unnecessary -> moved to bottom also
- [x] Manpage conventions
    - [x] Rename `USAGE AND OPTIONS` to `OPTIONS` -> done
    - [x] Replace tabular option documentation by
    ```
    --option
           Doc
    ````
    format -> reactivated unused filter
    - [x] ~~Lines are wide, but couldn't find a recommended line length~~ -> we're keeping it like this

### CODE REVIEW

**Comment by iw0nderhow** [comment]
I implemented the "move usage section around" and "remove unnecessary stuff" parts using some comment tags, I think it's easier to just use those instead of hard-coding certain section names or detecting the start of the TOC.

File: devscripts/prepare_manpage.py [hunk 1]
```diff
@@ -43,6 +48,46 @@ def main():
...
+    ret = []
+    usage_section = []
+    readme_without_usage = []
+    in_section = False
+    for line in readme.split('\n'):
+        if line.startswith('# '):
+            if line[2:].startswith('USAGE AND OPTIONS'):
+                in_section = True
+            else:
+                in_section = False
```
Comment (pukkandan) [inline devscripts/prepare_manpage.py:77]: same

File: devscripts/prepare_manpage.py [hunk 2]
```diff
@@ -43,6 +48,46 @@ def main():
...
+    EXCLUDED_SECTION_END_STRING = '<!-- END EXCLUDE FROM MANPAGE -->'
+    ret = []
+    in_section = False
+    for line in readme.split('\n'):
+        if EXCLUDED_SECTION_BEGIN_STRING in line:
+            in_section = True
+        if not in_section:
+            ret.append(line)
+        if EXCLUDED_SECTION_END_STRING in line:
+            in_section = False
```
Comment (pukkandan) [inline devscripts/prepare_manpage.py:62]: u can use regex to simplify this

File: devscripts/prepare_manpage.py [hunk 3]
```diff
@@ -43,6 +48,46 @@ def main():
...
+
+        if in_section:
+            usage_section.append(line)
+        else:
+            readme_without_usage.append(line)
+    for line in readme_without_usage:
+        if '<!-- MANPAGE: PUT USAGE HERE -->' in line:
+            ret.extend(usage_section)
+        else:
+            ret.append(line)
```
Comment (pukkandan) [inline devscripts/prepare_manpage.py:87]: same. just str.replace or re.sub can do this

**Review by pukkandan — CHANGES_REQUESTED** [review]
> Lines are wide, but couldn't find a recommended line length

Keep it wide imo

**Comment by iw0nderhow** [comment]
Thanks for all the suggestions, I think I've incorporated them all now. Sorry for force-pushing ^^

File: devscripts/prepare_manpage.py [hunk 4]
```diff
@@ -43,6 +49,33 @@ def main():
         outf.write(readme)
 
 
+def filter_excluded_sections(readme):
+    EXCLUDED_SECTION_BEGIN_STRING = '<!-- MANPAGE: BEGIN EXCLUDED SECTION -->'
+    EXCLUDED_SECTION_END_STRING = '<!-- MANPAGE: END EXCLUDED SECTION -->'
+    return re.sub(
+        '(?s)%s.+?%s\n' % (EXCLUDED_SECTION_BEGIN_STRING, EXCLUDED_SECTION_END_STRING),
```
Comment (pukkandan) [inline devscripts/prepare_manpage.py:56]:
```suggestion
        rf'(?s){re.escape(EXCLUDED_SECTION_BEGIN_STRING)}.+?{re.escape(EXCLUDED_SECTION_END_STRING)}\n'
```
do `re.escape` when putting literal string inside regex

**Comment by iw0nderhow** [comment]
Alright, think I'm done here for now :)

**Pull request merged**